### PR TITLE
Configurable pull request guide

### PR DESCRIPTION
**Pull request by jimis**
* Modified a bit the criteria for accepting pull requests
* Improved order and formatting
* Added new "Coding style" guidelines.

### CODE REVIEW

**Comment by jimis** [comment]
@kacfengine please have a look at the rendered output. Especially the last commit is adding new entries under "Coding style", I  can separate it in a different pull request if you find it controversial. My purpose is to improve code uniformity.

**Comment by jimis** [comment]
@kacfengine also take special notice of the new changelog section.

File: HACKING.md [hunk 1]
```diff
@@ -1,35 +1,47 @@
...
 
-Thanks for considering contributing to the CFEngine! We take pull-requests on
-GitHub at http://github.com/cfengine, and we have a public Redmine bug-tracker
-at http://bug.cfengine.com
+Thanks for considering contributing to CFEngine! We take pull-requests
+[on GitHub](https://github.com/cfengine/core/pulls) and we have a public
+[Redmine bug-tracker](https://dev.cfengine.com). Discussion is taking place
+on the [dev-cfengine](https://groups.google.com/forum/#!forum/dev-cfengine)
+and [help-cfengine](https://groups.google.com/forum/#!forum/help-cfengine)
+mailing lists.
```
Comment (nickanderson) [inline HACKING.md]: and IRC #cfengine and #cfengine-dev

File: HACKING.md [hunk 2]
```diff
@@ -228,15 +286,34 @@ macros in C code.
...
+    some system error occured that is worth logging to syslog.
+  * `LOG_LEVEL_INFO`: For cf-agent, changes that the agent performs
+    to the system, for example when a promise has been repaired. For
+    cf-serverd, `access_rules` denials for connected clients.
+  * `LOG_LEVEL_VERBOSE` :: Log *human readable* progress info useful to
+    users (i.e. sysadmins). Also errors that are unimportant or expected
+    in certain cases.
+  * `LOG_LEVEL_DEBUG`: Log anything else (for example various progress info).
+    Try to avoid "Entering function Foo()", but rather use for
+    "While copying, got reply '%s' from server".
```
Comment (kacfengine) [inline HACKING.md]: Should mention ``LOG_LEVEL_NOTICE`` as well. It's currently used mainly for files promises with change tracking.

**Comment by kacfengine** [comment]
+1

**Comment by jimis** [comment]
Thanks for the review, merging!

**Pull request merged**